api/include/opentelemetry/logs/logger.h

```diff
@@ -57,48 +56,72 @@ class Logger
   /**
    * The base Log(...) method that all other Log(...) overloaded methods will eventually call,
    * in order to create a log record.
-   *
-   * Note this takes in a KeyValueIterable for the resource and attributes fields.
    */
   virtual void Log(Severity severity,
                    nostd::string_view name,
                    nostd::string_view body,
-                   const common::KeyValueIterable &resource,
                    const common::KeyValueIterable &attributes,
                    trace::TraceId trace_id,
                    trace::SpanId span_id,
                    trace::TraceFlags trace_flags,
                    common::SystemTimestamp timestamp) noexcept = 0;
 
+  OPENTELEMETRY_DEPRECATED_MESSAGE("resource is removed and ignored")
+  void Log(Severity severity,
+           nostd::string_view name,
+           nostd::string_view body,
+           OPENTELEMETRY_MAYBE_UNUSED const common::KeyValueIterable &resource,
```

Comment:
I think it's helpful for users to have some time to upgrade the usage of Log API instead of just `compile error`.

Reply:
With logs in experimental, I think breaking changes should be acceptable and users should be prepared for it. This will keep the API much cleaner. We should definitely mention this as a breaking change in the Release description.

Reply:
OK, I can remove these later, should I keep `api/include/opentelemetry/common/macros.h` ? Maybe it's useful in the future?

Reply:
We can remove macros for now, can add them back later if required. Thanks.